core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -134,6 +138,15 @@ public JDBCUserStoreManager(RealmConfiguration realmConfig, int tenantId) throws
 //		}
 
         // new properties after carbon core 4.0.7 release.
+        Map<String, Object> userStorePropertiesMap = new HashMap<>(realmConfig.getUserStoreProperties());
+        String digestFunction = userStorePropertiesMap.get(JDBCRealmConstants.DIGEST_FUNCTION).toString();
```

Comment:
If `JDBCRealmConstants.DIGEST_FUNCTION` is not specified in the `userStorePropertiesMap`, won't this throw a NPE?

Reply:
If `digestFunction` is not specified in `userStorePropertiesMap`, it will be set to an empty string. So that, it will not throw NPE.

Reply:
Empty string will set only if we do by the UI right? if we are using APIs this will return a NULL right?
Anyway I think we need to check for blank